src/AppInstallerCommonCore/Manifest/ManifestValidation.cpp

```diff
@@ -234,4 +236,104 @@ namespace AppInstaller::Manifest
 
         return resultErrors;
     }
+
+    // Validate msix and msixbundle installer manifest
+    std::vector<ValidationError> ValidateMsixManifest(
+        const Msix::PackageVersion packageVersion,
+        const ManifestInstaller& installer,
+        Msix::MsixPackageManifestCache& msixManifestsCache,
+        bool treatErrorAsWarning)
+    {
+        std::vector<ValidationError> errors;
+        std::optional<Msix::OSVersion> installerMinOSVersion;
+
+        try
+        {
+            if (!installer.MinOSVersion.empty())
+            {
+                installerMinOSVersion = std::make_optional<Msix::OSVersion>(installer.MinOSVersion);
+            }
+        }
+        catch (const std::exception&)
+        {
+            errors.emplace_back(ManifestError::InvalidFieldValue, "MinimumOSVersion", installer.MinOSVersion);
+        }
+
+        std::vector<Msix::MsixPackageManifest> msixManifests;
+        try {
+            msixManifests = msixManifestsCache.GetAppPackageManifests(installer.Url);
+        }
+        catch (...)
+        {
+            errors.emplace_back(ManifestError::InstallerFailedToProcess, "InstallerUrl", installer.Url);
+        }
+
+        for (auto msixManifest : msixManifests)

```

Comment:
Actually, msixbundle has checks to make sure each payload msix within declare exactly the same core entries(including identity, package version, TDFs, etc) in the msix manifest